### PR TITLE
Transactions loading/syncronizing indicator and account page more accessible without node connection

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

Improve account page by adding indications of when transactions are loading/synchronizing, and making the account page work (in a limited sense) even when account has no `accountInfo` from node.

## Changes

- Ensure loading component takes priority over no transactions in list
- Added icon to show when account transactions are being synced with wallet proxy
- Made components on account page work with no account info
- Improved transaction loading logic (reducer)

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: app/features/TransactionSlice.ts [hunk 1]
```diff
@@ -330,17 +242,19 @@ export const updateTransactions = createAsyncThunk<
         }
 
         async function updateSubroutine(maxId: bigint) {
-            if (!account || controller.isAborted) {
-                controller.finish();
```
Comment (shjortConcordium) [inline app/features/TransactionSlice.ts:334]: Why have you removed the finish statements?

File: app/pages/Accounts/useAccountSync.ts [hunk 2]
```diff
@@ -71,36 +95,43 @@ export default function useAccountSync(): string | undefined {
...
-            dispatch(updateTransactions({ controller, onError: setError }));
+            dispatch(
+                updateTransactions({
+                    controller,
+                    onError: setError,
+                })
+            );
         }
+
+        dispatch(updateTransactions({ controller, onError: setError }));
```
Comment (shjortConcordium) [inline app/pages/Accounts/useAccountSync.ts:106]: Why have you added this?

File: app/pages/Accounts/useAccountSync.ts [hunk 3]
```diff
@@ -71,36 +95,43 @@ export default function useAccountSync(): string | undefined {
...
+
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [
-        account?.address,
-        accountInfo?.accountAmount,
-        account?.status,
-        controller.isAborted,
-    ]);
+    }, [account?.address, accountInfo?.accountAmount, account?.status]);
```
Comment (shjortConcordium) [inline app/pages/Accounts/useAccountSync.ts:109]: I might be misunderstanding what you doing, but the isAborted is in the deps, so the effect is re-run when the previous update has finished.

File: app/features/TransactionSlice.ts [hunk 4]
```diff
@@ -1,4 +1,11 @@
-import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import {
+    createAsyncThunk,
+    createSlice,
+    PayloadAction,
+    isAnyOf,
+    isFulfilled,
+} from '@reduxjs/toolkit';
+import { Mutex, MutexInterface } from 'async-mutex';
```
Comment (shjortConcordium) [inline app/features/TransactionSlice.ts:8]: You are using a new dependency, but you have not added it to package.json.

Comment (soerenbf) [reply app/features/TransactionSlice.ts:8]: weird.. just ran `yarn add` as usual... I'll try again

File: app/features/TransactionSlice.ts (hunk 1 above)
Comment (soerenbf) [reply app/features/TransactionSlice.ts:376]: Is moved to `extraReducers` in slice in most recent update

File: app/features/TransactionSlice.ts [hunk 5]
```diff
@@ -438,6 +341,172 @@ export const updateTransactions = createAsyncThunk<
...
+                    } else {
+                        state.transactions = action.payload.transactions;
+                    }
+                }
+
+                if (isLatest) {
+                    state.loadingTransactions = false;
+                }
+
+                controller?.finish();
```
Comment (shjortConcordium) [inline app/features/TransactionSlice.ts:417]: The controller on updateTransactions should only finish, when we have finished updating transactions (or it has been aborted). 
Currently it seems to finish when we load the first batch of transactions.

File: app/features/TransactionSlice.ts [hunk 6]
```diff
@@ -438,6 +341,172 @@ export const updateTransactions = createAsyncThunk<
...
+            }
+        );
+
+        builder.addMatcher(
+            isAnyOf(updateTransactions.rejected, updateTransactions.fulfilled),
+            (state, action) => {
+                state.synchronizing = false;
+                action.meta.arg.controller.finish();
```
Comment (shjortConcordium) [inline app/features/TransactionSlice.ts:425]: The controller on updateTransactions should only finish, when we have finished updating transactions (or it has been aborted).
Currently it seems to finish when we load the first batch of transactions.

Comment (shjortConcordium) [reply app/features/TransactionSlice.ts:425]: which also means it is not aborted, when you change the account / view

Comment (soerenbf) [reply app/features/TransactionSlice.ts:425]: hmm.. doesn't happen for me. The reject/fulfill of the action should happen when the update is recursively done, which happens when I run it.

Comment (shjortConcordium) [reply app/features/TransactionSlice.ts:425]: I'll show you tomorrow.

Comment (soerenbf) [reply app/features/TransactionSlice.ts]: This, along with the sync issue we talked about earlier today, should be fixed now.

**Comment by shjortConcordium** [comment]
If I have two accounts, and change the chosen account twice in quick succession, then the wallet will start spamming the wallet-proxy with request, even though it is caught up.

File: app/pages/Accounts/TransactionList/FiniteTransactionList.tsx [hunk 7]
```diff
@@ -27,7 +30,7 @@ export default function FiniteTransactionList({
 
     if (showLoading) {
         return (
-            <div className="flex">
+            <div className={clsx('flex', styles.thickBlueSeparatorTop)}>
                 <Loading
```
Comment (shjortConcordium) [inline app/pages/Accounts/TransactionList/FiniteTransactionList.tsx:34]: This Loading's text should be capitalized.

**Review by shjortConcordium — COMMENTED** [review]
If I change accounts multiple times, I sometimes end up in a state, where the transactions are never loaded, and the "loading transactions" are there permanently. (until I close the app)

**Comment by soerenbf** [comment]
> If I change accounts multiple times, I sometimes end up in a state, where the transactions are never loaded, and the "loading transactions" are there permanently. (until I close the app)

This should be fixed now.

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**